src/main/java/com/google/cloud/tools/internal/process/StreamConsumingProcessRunner.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.tools.internal.process;
+
+import com.google.cloud.tools.internal.process.io.StreamConsumer;
+
+import java.io.IOException;
+
+/**
+ * Process Runner impl that allows for specialized stream consumption
+ */
+public class StreamConsumingProcessRunner implements ProcessRunner {
+
+  private final StreamConsumer streamConsumer;
```

Comment:
What use cases will this class cover? I didn't find the need to read from streams while I was implementing the gcloud and dev_appserver interfaces.